File: FlinkJavaCode/src/main/java/com/wubaibao/flinkjava/code/chapter6/partitions/ReblanceTest.java
package com.wubaibao.flinkjava.code.chapter6.partitions;

import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * Flink Reblance 重平衡分区测试
 */
public class ReblanceTest {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);
        DataStreamSource<String> source = env.socketTextStream("node5", 9999);
        source.rebalance().print("shuffle").setParallelism(3);
        env.execute();
    }

}

File: FlinkScalaCode/src/main/scala/com/wubaibao/flinkscala/code/chapter6/partitions/ReblanceTest.scala
package com.wubaibao.flinkscala.code.chapter6.partitions

import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}

/**
 * Flink Reblance 重平衡分区测试
 */
object ReblanceTest {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)
    val ds: DataStream[String] = env.socketTextStream("node5", 9999)
    ds.rebalance.print().setParallelism(3)
    env.execute()
  }

}
